File: 백준/Gold/1956. 운동/운동.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

var (
	reader = bufio.NewReader(os.Stdin)
	writer = bufio.NewWriter(os.Stdout)
)

func main() {
	defer writer.Flush()
	var n, m, s, e int
	var c, ans float64
	fmt.Fscanf(reader, "%d %d\n", &n, &m)
	var roads = make([][]float64, n+1)
	for i := 0; i < n+1; i++ {
		roads[i] = make([]float64, n+1)
		for j := 0; j < n+1; j++ {
			roads[i][j] = math.MaxFloat64
		}
	}
	for i := 0; i < m; i++ {
		fmt.Fscanf(reader, "%d %d %f\n", &s, &e, &c)
		roads[s][e] = c
	}
	for i := 1; i < n+1; i++ {
		for j := 1; j < n+1; j++ {
			for k := 1; k < n+1; k++ {
				roads[j][k] = math.Min(roads[j][k], roads[j][i]+roads[i][k])
			}
		}
	}
	ans = math.MaxFloat64
	for i := 1; i < n+1; i++ {
		ans = math.Min(ans, roads[i][i])
	}
	if ans >= math.MaxFloat64 {
		ans = -1
	}
	fmt.Fprintf(writer, "%0.f", ans)
}

File: 백준/Gold/1956. 운동/운동.py
import sys 
input = sys.stdin.readline
n, m = map(int,input().split())
roads = [[float('inf')] * (n+1) for _ in range(n+1)]
for _ in range(m):
    s, e, c = map(int, input().split())
    roads[s][e] = min(roads[s][e], c)

for k in range(1, n + 1):
    for a in range(1, n + 1):
        for b in range(1, n + 1):
            roads[a][b] = min(roads[a][b], roads[a][k] + roads[k][b])
ans = float('inf')
for i in range(1, n + 1):
    ans = min(ans, roads[i][i])
if ans==float('inf'):
    print(-1)
else:
    print(ans)